ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml

```diff
@@ -6,16 +6,18 @@ ref:
     requests:
       cpu: 10m
       memory: 100Mi
+  credentials:
+  - namespace: test-credentials
+    name: cluster-secrets-openstack-vexxhost
+    mount_path: /var/run/cluster-secrets-openstack-vexxhost
+  - namespace: test-credentials
+    name: cluster-secrets-openstack-moc
+    mount_path: /var/run/cluster-secrets-openstack-moc
   env:
-    - name: CONFIG_TYPE
-    - name: CONFIG_API_VERSION
-    - name: OPENSTACK_EXTERNAL_NETWORK
-    - name: OPENSTACK_COMPUTE_FLAVOR
-    - name: OS_CLOUD
-    - name: BASE_DOMAIN
-    - name: KURYR_ENABLED
-      default: "false" 
+  - name: OS_CLIENT_CONFIG_FILE_VEXXHOST
+    default: '/var/run/cluster-secrets-openstack-vexxhost/clouds.yaml'
+  - name: OS_CLIENT_CONFIG_FILE_MOC
+    default: '/var/run/cluster-secrets-openstack-moc/clouds.yaml'
```

Comment:
> in the workflow.ref for the test I can write

Long story short: unless you mount all secrets in every step, you can't set an absolute path in `env`.

With this patch, cloud secrets are only mounted as a volume in one step. This means that if a job is compromised (which has happened more than once or twice recently), we would have better chances to only compromise one cloud.

**Cloud provider configuration is an implementation detail**; it has nothing to do with the test definition. Because of this, I don't think it should have a place in the workflow definition.

I also don't think it's worth optimising this particular step for adding a new cloud provider.

To add a new cloud provider, you have to:
* define a new lease and hook it to a CLUSTER_TYPE
* upload the clouds.yaml in the shiftstack-secrets namespace
* mirror the secret to a mountable volume
* finally, add the configuration to `ipi-conf-openstack`.

It's a number of tasks, mostly unrelated to this particular patch. And in my (biased) point of view, I think that this patch makes it clearer how to do it.